File: ClassicalDynamics/NewtonsLowsOfMotion/UniformMotion/Python/Unim1D.py
import math
import matplotlib.pyplot as plt
class Unim1D:
    #変数の初期化(constructor)
    def __init__(self, initX, initVX, deltaTime, totalTime):
        self.locationX = initX
        self.velocityX = initVX
        self.deltaTime = deltaTime
        self.totalTime = totalTime
        self.steps = int(self.totalTime / self.deltaTime)
    
    def calcX(self, nowTime : float) -> float:
        return self.locationX + self.velocityX * nowTime   

    def dataWrite(self):
        nowlocX, nowTime = 0, 0
        dataX, dataY = [], []
        for _count in range(self.steps):
            nowlocX = self.calcX(nowTime)
            #print(nowTime, nowlocX)
            dataX.append(nowTime)
            dataY.append(nowlocX)
            nowTime = nowTime + self.deltaTime
        
        plt.plot(dataX, dataY, 'go')
        plt.savefig("unim1D.png")


if __name__ == "__main__":
    initX = 0.0 # 座標xの初期値
    initVX = 2.0 #速度xの初期値
    deltaTime = 0.1 #経過時間の値
    totalTime = 4.0 #総合時間

    unim1d = Unim1D(initX, initVX, deltaTime, totalTime)
    unim1d.dataWrite()

File: ClassicalDynamics/NewtonsLowsOfMotion/UniformMotion/JavaScript/Unim1D.js
"use strict";
class Unim1D {
    constructor(initX, initVX, deltaTime, totalTime) {
        this.locationX = 0;
        this.velocityX = 0;
        this.deltaTime = 0;
        this.totalTime = 0;
        this.steps = 0;
        this.locationX = initX;
        this.velocityX = initVX;
        this.deltaTime = deltaTime;
        this.totalTime = totalTime;
        this.steps = Math.floor(this.totalTime / this.deltaTime);
    }
    calcX(nowTime) {
        return this.locationX + this.velocityX * nowTime;
    }
    dataWrite() {
        let nowlocX = 0, nowTime = 0;
        for (let count = 1; count <= this.steps; count++) {
            nowlocX = this.calcX(nowTime);
            console.log(nowTime, nowlocX);
            nowTime = nowTime + this.deltaTime;
        }
    }
}
let render = () => {
    let initX = 0.0; //座標xの初期値
    let initVX = 2.0; //速度xの初期値
    let deltaTime = 0.1; //経過時間の値
    let totalTime = 4.0; //総合時間
    let unim1d = new Unim1D(initX, initVX, deltaTime, totalTime);
    unim1d.dataWrite();
};
render();